sympy/core/power.py

```diff
@@ -1582,8 +1583,8 @@ def mul(d1, d2):
             # Convert floats like 0.5 to exact SymPy numbers like S.Half, to
             # prevent rounding errors which can induce wrong values of d leading
             # to a NotImplementedError being returned from the block below.
-            from sympy.simplify.simplify import nsimplify
-            _, d = nsimplify(g).leadterm(x, logx=logx)
+            g.replace(lambda x: x.is_Float and x.is_same(Rational(x), equal_valued), lambda x: Rational(x))
```

Comment:
This line computes an expression that is not assigned to anything.

Also I don't see why this complicated condition is needed. If `x` is a Float then `Rational(x)` will always be an exact conversion.

Reply:
> This line computes an expression that is not assigned to anything.

Really overlooked this from the start :), thanks for pointing out.